### PR TITLE
feat: add APIs to enable/disable spell checker

### PR DESCRIPTION
**Pull request by zcbenz**
#### Description of Change

Refs https://github.com/electron/electron/issues/3211.

This PR adds a `ses.spellCheckerEnabled` property which can be used to enable/disable the builtin spell checker, and a `toggleSpellChecker` menu item role that maps to the "Check Spelling While Typing" menu item.

Screen Shot 2020-11-05 at 9 30 38

To completely solve #3211 we also need some APIs for the spell panel, which are added in https://github.com/electron/electron/pull/26332.

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] tests are [changed or added](https://github.com/electron/electron/blob/master/docs/development/testing.md)
- [x] relevant documentation is changed or added
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: Add APIs to enable/disable spell checker.

### CODE REVIEW

File: docs/api/session.md [hunk 1]
```diff
@@ -802,6 +812,10 @@ The following properties are available on instances of `Session`:
 A `String[]` array which consists of all the known available spell checker languages.  Providing a language
 code to the `setSpellCheckerLanguages` API that isn't in this array will result in an error.
 
+#### `ses.spellCheckerEnabled`
+
+A `Boolean` indicating whther builtin spell checker is enabled.
```
Comment (dsanders11) [inline docs/api/session.md:817]: whthr => whether

File: typings/internal-electron.d.ts [hunk 2]
```diff
@@ -136,6 +136,7 @@ declare namespace Electron {
     overrideReadOnlyProperty(property: string, value: any): void;
     groupId: number;
     getDefaultRoleAccelerator(): Accelerator | undefined;
+    getCheckStatus(): boolean;
```
Comment (dsanders11) [inline typings/internal-electron.d.ts:139]: I think `getCheckStatus()` can return `undefined`?

Comment (zcbenz) [reply typings/internal-electron.d.ts:139]: The `roles.getCheckStatus()` can return `undefined`, but `MenuItem.getCheckStatus` can not, the code path that returns `undefined` should have been avoided.

**Review by miniak — APPROVED** [review]

**Comment by jkleinsc** [comment]
The API WG approved this PR at our Nov 9, 2020 meeting

**Review by jkleinsc — APPROVED** [review]

**Pull request merged**